userspace/libsinsp/event.cpp

```diff
@@ -2747,3 +2747,83 @@ uint64_t sinsp_evt::get_lastevent_ts() const
 {
 	return m_tinfo->m_lastevent_ts;
 }
+
+bool sinsp_evt::clone_event(sinsp_evt &dest, const sinsp_evt &src)
+{
+	dest.m_inspector = src.m_inspector;
+
+	if (src.m_pevt != nullptr)
+	{
+		dest.m_pevt_storage = new char[src.m_pevt->len];
+		memcpy(dest.m_pevt_storage, src.m_pevt, src.m_pevt->len);
+		dest.m_pevt = (scap_evt *) dest.m_pevt_storage;
+	}
+	else
+	{
+		dest.m_pevt_storage = nullptr;
+		dest.m_pevt = nullptr;
+	}
+
+	dest.m_poriginal_evt = nullptr;
+
+	// tinfo
+	if (src.m_tinfo_ref && src.m_tinfo && src.m_tinfo_ref.get() != src.m_tinfo)
+	{
+		// bad data
+		return false;
+	}
+
+	if (src.m_tinfo_ref)
+	{
+		dest.m_tinfo_ref = src.m_tinfo_ref;
+		dest.m_tinfo = dest.m_tinfo_ref.get();
+	}
+	else if (src.m_tinfo)
+	{
+		dest.m_tinfo_ref = dest.m_inspector->get_thread_ref(src.m_tinfo->m_tid, false, false);
+		if (dest.m_tinfo_ref == nullptr)
+		{
+			// no tinfo
+			return false;
+		}
+		dest.m_tinfo = dest.m_tinfo_ref.get();
+	}
+	else
+	{
+		dest.m_tinfo_ref = nullptr;
+		dest.m_tinfo = nullptr;
+	}
+
+	// scalars
+	dest.m_cpuid = src.m_cpuid;
```

Comment:
What i don't like about this solution: this is yet another point one has to remember to update when adding a new field to `sinsp_evt`.

Can we think of something different? This is just a nice to have :)

Reply:
what's the problem?

Reply:
btw when you have a copy/move ctor you have to think about all fields to populate correctly when you are adding one. It's not much diff from copy ctor - can be replaced with one that throws.